posts/2020/08/introducing-versioned-hdf5.md

```diff
@@ -0,0 +1,93 @@
+
+![Hierarchical Data Format (HDF5) Dataset (From https://www.neonscience.org/about-hdf5)](https://www.neonscience.org/sites/default/files/images/HDF5/hdf5_structure4.jpg)
+
+Since reading and writing operations in these large data files must be fast, the HDF5 model includes data compression and *chunking*. This technique allows the data to be retrieved in subsets that fit the computer's memory or RAM, which means that it doesn't require the entire file contents to be loaded into memory at once. All this makes HDF5 a popular format in several domains, and with [h5py](https://www.h5py.org) it is possible to use a Pythonic interface to read and write data to a HDF5 file. 
+
+Now, let's say you have an HDF5 file with contents that change over time. You may want to add or remove datasets, change the contents of the data or the metadata, and keep a record of which changes occurred when, with a way to recover previous versions of this file. Since HDF5 is a binary file format, using regular version control tools (such as git) may prove difficult. 
+
+Introducing the Versioned HDF5 library 
+--------------------------------------
+
+The Versioned HDF5 library is a versioned abstraction on top of h5py. Because of the flexibility of the HDF5 data model, all versioning data is stored in the file itself, which means that different versions of the same data (including version metadata) can be stored in a single HDF5 file. 
+
+To see how this works in practice, let's say we create a regular HDF5 file with h5py called `mydata.h5`.
+
+```py
+    >>> import h5py
+    >>> fileobject = h5py.File('mydata.h5', 'w')
+```
+
+Now, you can create a `VersionedHDF5file` object:
+
+```py
+    >>> from versioned_hdf5 import VersionedHDF5File
+    >>> versioned_file = VersionedHDF5File(fileobject)
+```
+
+This file still doesn't have any data or versions stored in it. To create a new version, you can use a context manager:
+
+```py
+    >>> with versioned_file.stage_version('version1') as group:
+    ...     group['mydataset'] = np.ones(10000)
+```
+
+The context manager returns a h5py group object, which should be modified in-place to build the new version. When the context manager exits, the version will be written to the file. This creates a `_version_data` group containing metadata associated with versions in the file:
+
```

Comment:
The `_version_data` group is an implementation detail. I would leave this out of the high level blog post. It should just mention here that you have to interact with the versioned groups and datasets using the versioned_hdf5 API, rather than trying to access the datasets using h5py.